File: src/main/java/leets/leenk/domain/birthday/domain/entity/BirthdayLetterReadMark.java
package leets.leenk.domain.birthday.domain.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "birthday_letter_read_marker")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class BirthdayLetterReadMark {
    @Id
    @Column(name = "receiver_id")
    private Long receiverId;

    @Column(name = "last_read_at", nullable = false)
    private LocalDateTime lastReadAt;

    public void markRead(LocalDateTime lastReadAt) {
        this.lastReadAt = lastReadAt;
    }
}

File: src/main/kotlin/leets/leenk/domain/birthday/domain/entity/BirthdayLetterReadMark.kt
package leets.leenk.domain.birthday.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "birthday_letter_read_marker")
class BirthdayLetterReadMark(
    @Id
    @Column(name = "receiver_id")
    val receiverId: Long,
    @Column(name = "last_read_at", nullable = false)
    var lastReadAt: LocalDateTime,
) {
    fun markRead(lastReadAt: LocalDateTime) {
        this.lastReadAt = lastReadAt
    }
}
